src/nodeProvider.ts

```diff
@@ -22,10 +22,12 @@ export class DidactNodeProvider implements vscode.TreeDataProvider<TreeNode> {
 	private _onDidChangeTreeData: vscode.EventEmitter<TreeNode | undefined> = new vscode.EventEmitter<TreeNode | undefined>();
 	readonly onDidChangeTreeData: vscode.Event<TreeNode | undefined> = this._onDidChangeTreeData.event;
 
-	protected treeNodes: TreeNode[] = [];
-	protected retrieveTutorials : boolean = true;
+	protected treeNodes: TreeNode[];
+	protected retrieveTutorials = true;
 
-	constructor() {}
+	constructor() {
+		this.treeNodes = [];
```

Comment:
what's the advantage of initializing in the constructor?

Reply:
I had to have something in the constructor, otherwise ESLint would complain about it. So I decided to move the least risky of the initializations to the constructor.

Reply:
Can you point to the rule that was broken?

Reply:
![image](https://user-images.githubusercontent.com/435853/93449248-d1830000-f8d4-11ea-8bfb-274868807c21.png)

Reply:
https://eslint.org/docs/rules/no-empty-function
based on examples, it is possible to specify a comment to say that the constructor can be empty.
otherwise, why the constructor is needed if there is nothing inside? Can it be removed?

Reply:
Having no constructor at all seems to work too.